File: 1956-maximum-element-after-decreasing-and-rearranging/maximum-element-after-decreasing-and-rearranging.py
class Solution:
    def maximumElementAfterDecrementingAndRearranging(self, arr: List[int]) -> int:
        arr.sort()
        # first element should be 1
        # check the second condition for all the elements
        arr[0] = 1
        for i in range(1, len(arr)):
            if arr[i] - arr[i - 1] > 1:
                arr[i] = arr[i - 1] + 1

        return arr[-1]

File: 1956-maximum-element-after-decreasing-and-rearranging/maximum-element-after-decreasing-and-rearranging.java
/**
    1   1   2
 */


class Solution {
    public int maximumElementAfterDecrementingAndRearranging(int[] arr) {
        // what the element should be,
        // what it actually is
        Arrays.sort(arr);
        int expected = 1;
        for (int num : arr) {
            if (num >= expected) {
                expected += 1;
            } else {
                expected = num + 1;
            }
        }
        return expected - 1;
    }
}